### PR TITLE
Fix truncation while reading strings with spaces

### PR DESCRIPTION
**Pull request by jayeshkrishna**
Fixing issue with truncation while writing/reading strings with
spaces in a variable using pio_put/get_var.

Also modifying existing tests and adding some new tests that
show the issue with reading strings with spaces.

Also see E3SM-Project/E3SM#3966

### CODE REVIEW

**Comment by jayeshkrishna** [comment]
This PR development is done and the issues are resolved. I am in the process of doing the final round of testing.

File: src/flib/pionfput_mod.F90.in [hunk 1]
```diff
@@ -111,22 +111,48 @@ contains
...
+    if(ierr /= PIO_NOERR) then
+      call piodie(__PIO_FILE__, __LINE__, "Getting text variable size failed")
+    end if
+
+    if(clen < len_trim(ival)) then
+      print *, "PIO: WARNING: The length of the string in the user buffer,",&
+                len_trim(ival),&
+                " chars, is greater than the size of the variable",&
+                " being written to, ", clen, " chars. The string",&
+                " may be truncated"
```
Comment (dqwu) [inline src/flib/pionfput_mod.F90.in:132]: We can add file and line info for this warning:
`" may be truncated" ==> " may be truncated. ", __PIO_FILE__, __LINE__`

**Review by dqwu — APPROVED** [review]

**Comment by rljacob** [comment]
When will this be in E3SM master?

**Comment by tktesfa2014** [comment]
I have a PR waiting to be merged. I expect it be merged sometime next month.
Thanks,
Teklu

**Comment by tktesfa2014** [comment]
Sorry, Rob.
Please ignore my previous response. It was intended for another PR.
Thanks,
Teklu

**Comment by jayeshkrishna** [comment]
@rljacob , We are cutting a Scorpio release today (I also wanted to include the fix for the assertions that @ambrad saw in his debug runs) and I will have an E3SM PR today for bringing these changes to E3SM master.

**Comment by jayeshkrishna** [comment]
The E3SM PR https://github.com/E3SM-Project/E3SM/pull/3992 should bring this change to E3SM master.

**Pull request merged**